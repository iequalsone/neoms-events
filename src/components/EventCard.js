import React, { Component } from "react";
import { NavLink } from "react-router-dom";
import TagLink from "./TagLink";
import dateFormat from "dateformat";
import placeholderImage from "../images/events-placeholder.jpg";

class EventCard extends Component {
  renderImage(image, member_profile_image, title) {
    // console.log(image);
    // console.log(member_profile_image);
    if (!image) {
      // console.log(member_profile_image);
      if (!member_profile_image) {
        return (
          <img className="card-img-top" src={placeholderImage} alt={title} />
        );
      }
      return (
        <img className="card-img-top" src={member_profile_image} alt={title} />
      );
    }

    return <img className="card-img-top" src={image[0]} alt={title} />;
  }

  renderDateTime(start_date) {
    // console.log(start_date);
    if (typeof start_date !== "undefined") {
      return (
        <p className="card-text start-date">
          {dateFormat(start_date, "mmmm d, yyyy, h:MM TT")}
        </p>
      );
    }

    return;
  }

  renderTags(tags) {
    if (typeof tags !== "undefined") {
      let output = tags.map(t => {
        return <TagLink key={t} tag={t} />;
      });

      return output;
    }
  }

  renderFloatingTag(floatingTag) {
    if (floatingTag === "") {
      return;
    }

    return <div className="floating-tag">{floatingTag}</div>;
  }

  renderRows() {
    // console.log(this.props);
    const {
      slug,
      image,
      member_profile_image,
      title,
      start_datetime,
      location,
      tags,
      floatingTag,
    } = this.props.event;
    var path = "/" + process.env.REACT_APP_SINGLE_URL + "/" + slug;

    return (
      <div className="col-xs-12 col-sm-6 col-md-6 col-lg-4">
        <div className="card mb-4 box-shadow">
          {this.renderFloatingTag(floatingTag)}
          <NavLink to={path}>{this.renderImage(image, member_profile_image, title)}</NavLink>
          <div className="card-body">
            <NavLink to={path}>
              {this.renderDateTime(start_datetime)}
              <h3 className="card-text title">{title}</h3>
              <p className="card-text location">{location}</p>
            </NavLink>
            <div className="justify-content-between align-items-center">
              <p className="card-text tags">{this.renderTags(tags)}</p>
            </div>
          </div>
        </div>
      </div>
    );
  }

  // componentDidMount() {
  //   // console.log(this.props);
  // }

  render() {
    return this.renderRows();
  }
}

export default EventCard;
