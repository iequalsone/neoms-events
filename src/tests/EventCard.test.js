import React from 'react';
import EventCard from '../components/EventCard';
import { configure, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const event = {
    slug: 'slug',
    image: 'image.jpg',
    member_profile_image: 'profile.jpg',
    title: 'Title',
    start_datetime: '00223232',
    location: 'Somewhere',
    tags: { map: () => { } },
    floatingTag: 'Floating Tag',
  }

  const component = shallow(
    <Router>
      <EventCard event={event} />
    </Router>
  );
  expect(component).toMatchSnapshot();
});