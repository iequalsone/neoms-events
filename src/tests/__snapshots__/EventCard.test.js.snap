// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserRouter>
    <EventCard
        event={
            Object {
                "floatingTag": "Floating Tag",
                "image": "image.jpg",
                "location": "Somewhere",
                "member_profile_image": "profile.jpg",
                "slug": "slug",
                "start_datetime": "00223232",
                "tags": Object {
                  "map": [Function],
                },
                "title": "Title",
              }
        }
    />
</BrowserRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <EventCard
        event={
                Object {
                        "floatingTag": "Floating Tag",
                        "image": "image.jpg",
                        "location": "Somewhere",
                        "member_profile_image": "profile.jpg",
                        "slug": "slug",
                        "start_datetime": "00223232",
                        "tags": Object {
                          "map": [Function],
                        },
                        "title": "Title",
                      }
        }
/>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "event": Object {
          "floatingTag": "Floating Tag",
          "image": "image.jpg",
          "location": "Somewhere",
          "member_profile_image": "profile.jpg",
          "slug": "slug",
          "start_datetime": "00223232",
          "tags": Object {
            "map": [Function],
          },
          "title": "Title",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <EventCard
          event={
                    Object {
                              "floatingTag": "Floating Tag",
                              "image": "image.jpg",
                              "location": "Somewhere",
                              "member_profile_image": "profile.jpg",
                              "slug": "slug",
                              "start_datetime": "00223232",
                              "tags": Object {
                                "map": [Function],
                              },
                              "title": "Title",
                            }
          }
/>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "event": Object {
            "floatingTag": "Floating Tag",
            "image": "image.jpg",
            "location": "Somewhere",
            "member_profile_image": "profile.jpg",
            "slug": "slug",
            "start_datetime": "00223232",
            "tags": Object {
              "map": [Function],
            },
            "title": "Title",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
